integrationTests/src/test/java/org/sonarqube/gradle/AndroidTest.java

```diff
@@ -50,11 +50,11 @@ public void testAndroidProjectJdk8Retrolambda() throws Exception {
         baseDir.resolve("src/main/res"),
         baseDir.resolve("src/main/AndroidManifest.xml"));
     assertThat(Paths.get(props.getProperty("sonar.tests"))).isEqualTo(baseDir.resolve("src/test/java"));
-    assertThat(Paths.get(props.getProperty("sonar.java.binaries"))).isEqualTo(baseDir.resolve("build/intermediates/javac/debug/compileDebugJavaWithJavac/classes"));
+    assertThat(Paths.get(props.getProperty("sonar.java.binaries"))).isEqualTo(baseDir.resolve("build/intermediates/javac/debug/classes"));
```

Comment:
All those changes are probably related to the update of the Android Gradle plugin to 4.2.0, but since we are running tests in a matrix, you have to use a condition to still support older versions.

Reply:
ok I got this, let me try a more clean solution to this

Reply:
maybe JUnit's `assume` mechanism will help us in conditional test running, but fixed versions for conditional switching shouldn't be a clean solution, cannot find out any document or source code that make decision on where to put the intermediates, any suggestion @henryju ?

Reply:
@henryju  Yes I will update the CI script, the problem is maybe some old Gradle versions are not able to build dynamic features module, in such a case I intend to:

- Revert `AndroidTestingBlueprint` to original state
- Clone another blue print project which has dynamic feature

After that, the old test cases will skip on new Gradle version, and the new test cases will run only new Gradle version. How do you think?

Reply:
That sound very good to me :)

Reply:
@henryju I updated test cases with some conditional checks as discussed, could you please take another review?